### PR TITLE
[PDI-8928]  KettleDatabaseRepositoryConnectionDelegate - multiple un-…

### PR DESCRIPTION
**Pull request by AliaksandrShuhayeu**
[PDI-8928]  KettleDatabaseRepositoryConnectionDelegate - multiple unsynchronized methods that mutate objects.
extracted code for modifying sqlMap in separated synchronized method

### CODE REVIEW

**Comment by AliaksandrShuhayeu** [comment]
@mbatchelor Could you please review it?

**Comment by AliaksandrShuhayeu** [comment]
@lucboudreau Could you please review it?
Fails of wingman  don't applies to my changes.

File: engine/src/org/pentaho/di/repository/kdr/delegates/KettleDatabaseRepositoryConnectionDelegate.java [hunk 1]
```diff
@@ -1917,6 +1897,15 @@ public synchronized void insertTableRow( String tablename, RowMetaAndData values
     }
   }
 
+  private synchronized PreparedStatement getPreparedStatement( String sql ) throws KettleDatabaseException {
```
Comment (mbatchelor) [inline engine/src/org/pentaho/di/repository/kdr/delegates/KettleDatabaseRepositoryConnectionDelegate.java]: Even though there’s no time now to completely rewrite this tangled mass of code, we can’t just add another issue to an already bad problem. It has to start somewhere, and this is where. Please re-work your fix as follows:
•	Change sqlMap from a HashMap to a ConcurrentHashMap
•	Remove the synchronized keyword from your new method.
•	You should use the putIfAbsent method to add to the map

**Comment by AliaksandrShuhayeu** [comment]
Code issue is fixed.
@mbatchelor Could you please review it?

**Pull request merged**